target/linux/ath79/dts/qca955x.dtsi

```diff
@@ -379,4 +381,6 @@
 
 	resets = <&rst 13>, <&rst 23>;
 	reset-names = "mac", "mdio";
+	clocks = <&pll ATH79_CLK_AHB>, <&pll ATH79_CLK_REF>;
```

Comment:
this is interesting.

Is this compatible with the downstream driver as well?

https://github.com/openwrt/openwrt/commit/32726846c85985fbc320d123a7b26b32124cd47a says something about the upstream driver not supporting clocks. I assume this means these sections would be ignored with the downstream driver.

I ask as it would be interesting to switch to the upstream driver in one PR and DSA module conversion in another. OTOH that might not really make sense. I don't even know if the upstream driver works with swconfig.

Reply:
The dowstream driver sets the clocks directly, so it does not use the values ​​of DTS.

The upstream clock driver does not support Ethernet or MDIO clocks at all. The current upstream ag71xx driver only enables the ethernet clock and does not configure it. The MDIO clock is only used to calculate the clock divider of the MDIO bus clock.

The Ethernet clock can be configured to 100Mhz or 125Mhz and the MDIO clock can be REF, 50Mhz or 100Mhz. These bits are set in the bootloader, so there is no need to change them on the linux side. This is why setting the ethernet clock to only something that is already in use and the MDIO clock to REF works without proper clock support.

I haven't tested if upstream driver works with swconfig